pkg/service/service.go

```diff
@@ -262,8 +267,10 @@ func (s *Service) UpsertService(
 	}
 
 	localBackendCount := len(backendsCopy)
-	s.healthServer.UpsertService(lb.ID(svc.frontend.ID), svc.svcNamespace, svc.svcName,
-		localBackendCount, svc.svcHealthCheckNodePort)
+	if option.Config.EnableHealthCheckNodePort {
```

Comment:
Line 768 also needs a conditional branch such as this one

Reply:
Done